lib/elixir/lib/access.ex

```diff
@@ -360,6 +382,11 @@ defmodule Access do
   end
 
   def pop(nil, key) do
+    pop_raise(key)
+  end
+
+  @doc false
+  def pop_raise(key) do
```

Comment:
If we are going to do this, I would rather `raise` directly on Kernel. Please keep it as it was or raise on kernel.